torch/cuda/random.py

```diff
@@ -49,21 +44,33 @@ def set_rng_state(new_state: Tensor, device: Union[int, str, torch.device] = 'cu
             Default: ``'cuda'`` (i.e., ``torch.device('cuda')``, the current CUDA device).
     """
     new_state_copy = new_state.clone(memory_format=torch.contiguous_format)
-    if isinstance(device, str):
-        device = torch.device(device)
-    elif isinstance(device, int):
-        device = torch.device('cuda', device)
+    final_device = _get_device(device)
 
     def cb():
-        idx = cast(torch.device, device).index
-        if idx is None:
-            idx = current_device()
-        default_generator = torch.cuda.default_generators[idx]
+        default_generator = _get_generator(final_device)
         default_generator.set_state(new_state_copy)
 
     _lazy_call(cb)
 
 
+def set_rng_state_offset(offset: int, device: Union[int, str, torch.device] = 'cuda') -> None:
```

Comment:
You're publishing this as public API but it's not entirely clear this makes sense as public API? Some questions that would at least have to be documented here: When do I use `set_rng_state_offset` versus `set_rng_state`? What's the difference? Can I use these two APIs together? Do I get deterministic results if I reset two distinct RNGs to the same offset?

If we you don't want to upfront answer these questions, it would seem better to NOT make this public API to start.